includes/language/MessageCache.php

```diff
@@ -1030,7 +1030,11 @@ public function get( $key, $useDB = true, $langcode = true ) {
 		// Normalise title-case input (with some inlining)
 		$lckey = self::normalizeKey( $key );
 
-		$this->hookRunner->onMessageCache__get( $lckey );
+		// Fandom change: Workaround T193271 performance regression by letting extensions signal
+		// that a particular message key cannot exist in the database
+		if ( !$this->hookRunner->onMessageCache__get( $lckey ) ) {
+			return false;
+		}
```

Comment:
TODO: missing code in mw143: https://gerrit.wikimedia.org/r/q/T193271
Is it still required?